node/src/accountant/mod.rs

```diff
@@ -939,281 +630,162 @@ impl Accountant {
-            ScanType::Receivables => self
-                .scanners
-                .receivables
-                .scan(self, Some(response_skeleton)),
-            ScanType::PendingPayables => self
-                .scanners
-                .pending_payables
-                .scan(self, Some(response_skeleton)),
-        }
-    }
-
-    fn handle_cancel_pending_transaction(&self, msg: CancelFailedPendingTransaction) {
-        match self
-            .pending_payable_dao
-            .mark_failure(msg.id.rowid)
-        {
-            Ok(_) => warning!(
-                self.logger,
-                "Broken transaction {:?} left with an error mark; you should take over the care of this transaction to make sure your debts will be paid because there is no automated process that can fix this without you", msg.id.hash),
-            Err(e) => panic!("Unsuccessful attempt for transaction {:?} to mark fatal error at payable fingerprint due to {:?}; database unreliable", msg.id.hash,e),
-        }
-    }
-
-    fn handle_confirm_pending_transaction(&mut self, msg: ConfirmPendingTransaction) {
-        if let Err(e) = self
-            .payable_dao
-            .transaction_confirmed(&msg.pending_payable_fingerprint)
-        {
-            panic!(
-                "Was unable to uncheck pending payable '{:?}' after confirmation due to '{:?}'",
-                msg.pending_payable_fingerprint.hash, e
-            )
-        } else {
-            self.financial_statistics.total_paid_payable += msg.pending_payable_fingerprint.amount;
-            debug!(
-                self.logger,
-                "Confirmation of transaction {:?}; record for payable was modified",
-                msg.pending_payable_fingerprint.hash
-            );
-            if let Err(e) = self.pending_payable_dao.delete_fingerprint(
-                msg.pending_payable_fingerprint
-                    .rowid_opt
-                    .expectv("initialized rowid"),
-            ) {
-                panic!("Was unable to delete payable fingerprint '{:?}' after successful transaction due to '{:?}'",msg.pending_payable_fingerprint.hash,e)
-            } else {
+    fn handle_scan_for_payable_request(&mut self, response_skeleton_opt: Option<ResponseSkeleton>) {
+        match self.scanners.payable.begin_scan(
+            SystemTime::now(),
+            response_skeleton_opt,
+            &self.logger,
+        ) {
+            Ok(message) => {
+                self.report_accounts_payable_sub_opt
+                    .as_ref()
+                    .expect("BlockchainBridge is unbound")
+                    .try_send(message.clone())
+                    .expect("BlockchainBridge is dead");
+                eprintln!("Message was sent to the blockchain bridge, {:?}", message);
+            }
+            Err(BeginScanError::CalledFromNullScanner) => {
+                if cfg!(test) {
+                    eprintln!("Payable scan is disabled.");
```

Comment:
I have specified that in the enum variant as a comment. Using a log here would mean that these logs will only appear during the tests. This will bring confusion. So, I'd rather prefer code comments here.